samples/apps/copilot-chat-app/webapi/SemanticKernelExtensions.cs

```diff
@@ -106,7 +106,7 @@ private static void AddSemanticTextMemory(this IServiceCollection services)
                 services.AddSingleton<IMemoryStore, VolatileMemoryStore>();
                 services.AddScoped<ISemanticTextMemory>(sp => new SemanticTextMemory(
                     sp.GetRequiredService<IMemoryStore>(),
-                    sp.GetRequiredService<IOptions<AIServiceOptions>>().Value
+                    (ITextEmbeddingGeneration)sp.GetRequiredService<IOptions<AIServiceOptions>>().Value
```

Comment:
Maybe I'm missing something here, but it seems to me AIServiceOptions doesn't implement ITextEmbeddingGeneration.

Isn't this gonna break?

Reply:
It's casting the result of ToTextEmbeddingsService, not GetRequiredService, which has a return type of IEmbeddingGeneration and ITextEmbeddingGeneration extends IEmbeddingGeneration.

![image](https://github.com/microsoft/semantic-kernel/assets/125500434/715f2822-29e8-42d8-a5ed-b1b34601c365)

This was the original error: 
![image](https://github.com/microsoft/semantic-kernel/assets/125500434/8615b28b-02d8-4c62-a764-0bdd53ec7239)

Reply:
Oh yeah, reviewed too quick and didn't see the next line :)